openmrs/src/main/java/org/motechproject/openmrs/service/impl/OpenMRSEncounterServiceImpl.java

```diff
@@ -58,8 +59,7 @@ public Encounter createEncounter(String configName, Encounter encounter) {
 
             eventRelay.sendEventMessage(new MotechEvent(EventKeys.CREATED_NEW_ENCOUNTER_SUBJECT, EventHelper.encounterParameters(createdEncounter)));
         } catch (HttpClientErrorException e) {
-            LOGGER.error("Could not create encounter: " + e.getMessage());
-            return null;
+            throw new OpenMRSException("Could not create encounter with patient uuid: " + encounter.getPatient().getUuid() + " " + e.getMessage() + " " + e.getResponseBodyAsString(), e);
```

Comment:
please use String.format. We should avoid concatting strings, because it creates and stores multiple strings in Java String pool